src/protobuf-net/Serializers/ArrayDecorator.cs

```diff
@@ -116,6 +116,19 @@ protected override void EmitWrite(ProtoBuf.Compiler.CompilerContext ctx, ProtoBu
                             ctx.EmitCall(mappedWriter.GetMethod("EndSubItem"));
                         }
                     }
+                    else
+                    {
+                        Compiler.CodeLabel allDone = ctx.DefineLabel();
+                        ctx.LoadLength(arr, false);
+                        ctx.LoadValue(0);
+                        ctx.BranchIfGreater(allDone, false);
+
+                        ctx.LoadValue(fieldNumber);
+                        ctx.LoadReaderWriter();
+                        ctx.EmitCall(ctx.MapType(typeof(ProtoWriter)).GetMethod(nameof(ProtoWriter.WriteEmptyList)));

```

Comment:
it seems like this is going to change the wire format. will the existing version of protobuf work if this is there? Is there a way we can test this?

Reply:
So there isn't a way we can test this with unit tests in this repo directly. You could write a couple of small apps - one that serialized a bunch of messages with the new version and one that deserialized them with the old.

In terms of format change - it just adds a new field that has a very high field number to represent empty list. So older versions see that as 'I am deserializing a message with an unknown field' and skip past it.